firebase-storage/ktx/src/main/kotlin/com/google/firebase/storage/ktx/Storage.kt

```diff
@@ -45,6 +50,90 @@ fun storageMetadata(init: StorageMetadata.Builder.() -> Unit): StorageMetadata {
     return builder.build()
 }
 
+/**
+ * Destructuring declaration for [UploadTask.TaskSnapshot] to provide bytesTransferred
+ *
+ * @return the bytesTransferred of the [UploadTask.TaskSnapshot]
+ */
+operator fun UploadTask.TaskSnapshot.component1() = bytesTransferred
+
+/**
+ * Destructuring declaration for [UploadTask.TaskSnapshot] to provide totalByteCount
+ *
+ * @return the totalByteCount of the [UploadTask.TaskSnapshot]
+ */
+operator fun UploadTask.TaskSnapshot.component2() = totalByteCount
+
+/**
+ * Destructuring declaration for [UploadTask.TaskSnapshot] to provide its metadata
+ *
+ * @return the metadata of the [UploadTask.TaskSnapshot]
+ */
+operator fun UploadTask.TaskSnapshot.component3() = metadata
+
+/**
+ * Destructuring declaration for [UploadTask.TaskSnapshot] to provide its uploadSessionUri
+ *
+ * @return the uploadSessionUri of the [UploadTask.TaskSnapshot]
+ */
+operator fun UploadTask.TaskSnapshot.component4() = uploadSessionUri
+
+/**
+ * Destructuring declaration for [StreamDownloadTask.TaskSnapshot] to provide bytesTransferred
+ *
+ * @return the bytesTransferred of the [StreamDownloadTask.TaskSnapshot]
+ */
+operator fun StreamDownloadTask.TaskSnapshot.component1() = bytesTransferred
+
+/**
+ * Destructuring declaration for [StreamDownloadTask.TaskSnapshot] to provide totalByteCount
+ *
+ * @return the totalByteCount of the [StreamDownloadTask.TaskSnapshot]
+ */
+operator fun StreamDownloadTask.TaskSnapshot.component2() = totalByteCount
+
+/**
+ * Destructuring declaration for [StreamDownloadTask.TaskSnapshot] to provide its stream
+ *
+ * @return the stream of the [StreamDownloadTask.TaskSnapshot]
+ */
+operator fun StreamDownloadTask.TaskSnapshot.component3() = stream
+
+/**
+ * Destructuring declaration for [FileDownloadTask.TaskSnapshot] to provide bytesTransferred
+ *
+ * @return the bytesTransferred of the [FileDownloadTask.TaskSnapshot]
+ */
+operator fun FileDownloadTask.TaskSnapshot.component1() = bytesTransferred
+
+/**
+ * Destructuring declaration for [FileDownloadTask.TaskSnapshot] to provide totalByteCount
+ *
+ * @return the totalByteCount of the [FileDownloadTask.TaskSnapshot]
+ */
+operator fun FileDownloadTask.TaskSnapshot.component2() = totalByteCount
+
+/**
+ * Destructuring declaration for [ListResult] to provide its items
+ *
+ * @return the items of the [ListResult]
+ */
+operator fun ListResult.component1(): MutableList<StorageReference> = items
```

Comment:
Should the developers be able to mutate the list or can we return just `List<StorageReference>` here?
cc @schmidt-sebastian

Reply:
@vkryachko Seems like you're right. The java code does return an immutable list:

https://github.com/firebase/firebase-android-sdk/blob/334aa44e0f7a3418dbe5cb0da3dcc0abd442c967/firebase-storage/src/main/java/com/google/firebase/storage/ListResult.java#L86-L89

The `MutableList` return type was  mistakenly added by the IDE when auto-fixing this lint warning:

![Screenshot from 2020-07-13 17-22-57](https://user-images.githubusercontent.com/16766726/87322551-d8387400-c52d-11ea-91b5-c1e0bf517688.png)

Reply:
`List<StorageReference>` is the correct return type.

Reply:
Thanks for the confirmation @schmidt-sebastian 

Pushed a new commit with the update.